libtransmission/announcer.cc

```diff
@@ -295,21 +295,82 @@ struct tr_tracker
         }
     }
 
+    [[nodiscard]] constexpr auto seeder_count() const noexcept
+    {
+        return seeder_count_;
+    }
+
+    constexpr bool set_seeder_count(std::optional<int64_t> seeder_count_in) noexcept
+    {
+        if (seeder_count_in >= 0)
+        {
+            seeder_count_ = seeder_count_in;
+            return true;
+        }
+        return false;
+    }
+
+    [[nodiscard]] constexpr auto leecher_count() const noexcept
+    {
+        return leecher_count_;
+    }
+
+    constexpr bool set_leecher_count(std::optional<int64_t> leecher_count_in) noexcept
+    {
+        if (leecher_count_in >= 0)
+        {
+            leecher_count_ = leecher_count_in;
+            return true;
+        }
+        return false;
+    }
+
+    [[nodiscard]] constexpr auto download_count() const noexcept
+    {
+        return download_count_;
+    }
+
+    constexpr bool set_download_count(std::optional<int64_t> download_count_in) noexcept
+    {
+        if (download_count_in >= 0)
```

Comment:
Ah this is nice; TIL you can compare an optional to a value!

> Compares opt with a value. The values are compared (using the corresponding operator of T) only if opt contains a value. Otherwise, opt is considered less than value.